### ISSUE TITLE
Characters (such as "τ" or "π") not displaying correctly with Processing 3.3.7 build 0264 (latest)

### ISSUE THREAD

**Issue by glvjr** [processing/processing#5413]
## Description
Characters such as "τ" or "π"  are replaced with a "?" when displaying text to screen in Processing 3.3.7 build 0264

## Expected Behavior
These characters should display correctly in latest  Processing 3.3.7 build 0264

## Current Behavior
 They did display correctly in Processing 3.3.6 build 0263

## Steps to Reproduce
Run this code:
fill(0);
text("τ π", 50, 50);
1. With Processing 3.3.6 build 0263 which displays correctly.
2 . With Processing 3.3.7 build 0264 which replaces characters with ?.

Latest build:
https://github.com/processing/processing
Build instructions:
//https://github.com/processing/processing/wiki/Build-Instructions

## Your Environment

 

* Processing version: 3.3.6 build 0263 (works) and 3.3.7 build 0264 (issues with fonts)
* Operating System and OS version: W10
* Other information:

**Comment by glvjr** [comment]
Hello,

I did another build with "0264" from https://github.com/processing/processing and a screen grab of result.
![image](https://user-images.githubusercontent.com/11392209/36402601-c0b068ba-15ad-11e8-8530-2cd1e419921d.png)

**Comment by benfry** [comment]
Looks like the newest version of Java may be making default font worse. Not much I can do about it; you're better off setting a font explicitly, and making sure it includes the characters you want.

**Comment by benfry** [comment]
Sorry, was totally wrong on this one. It's actually a regression caused by [this change](https://github.com/processing/processing/pull/5152). Oddly, it's only a problem on Windows.

**Comment by benfry** [comment]
The workaround in the meantime is to use Unicode escapes in code… i.e. `\u03C0` for π.

**Comment by JakubValtar** [comment]
@benfry I can't reproduce it on my Windows 10. With or without Unicode substitution, it renders fine. How did you confirm that it is caused by that pull request?

**Comment by JakubValtar** [comment]
@benfry Ok I think I have it. Looks like JavaBuild write the files with default system charset instead of unicode. That explains why substituing unicode fixes it and also why the only affected platform is Windows. Fix coming soon.

**Comment by JakubValtar** [comment]
@benfry This should do the trick.

**Comment by benfry** [comment]
@JakubValtar Can't believe it wasn't being written in UTF-8, though that may date to early 2000s compiler weirdness with jikes and others (that's where the preproc option came from, at any rate).

@glvjr Thanks for catching this in the build / sorry for the earlier mis-diagnosis.

**Comment by JakubValtar** [comment]
@benfry We should check the whole codebase later, I ran "Implicit usage of platform's default charset" inspection and got a pile of warnings (some may be valid uses, some not)

**Comment by glvjr** [comment]
Thanks for this fix! 

On a positive note this was an opportunity for me to learn so much about fonts (coded a font viewer), the Processing source code and modifying source and compiling a build!

Note:
The console does not print the characters and shows" ? ?"
print("τ π");
Perhaps this is related to some of the discussions above.
Should I submit this as an "issue"

**Comment by benfry** [comment]
Yeah, that looks like a new issue. Though if you've changed the default font being used by the editor/console then you might be using a typeface that doesn't support those characters.
